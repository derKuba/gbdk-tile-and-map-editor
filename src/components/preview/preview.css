:host {
  display: inline-flex;
}

.preview-box {
  margin-left: 50px;
  display: flex;
  flex-direction: column;
  border: 1px solid black;
}


.preview-box.standalone {
  margin: 0;
  border: 0;

}

.preview-box.small {
  margin-top: -8px;
  display: inline-flex;
}

.matrixRow {
  display: inline-flex;
}

.preview-tile.large {
  width: 18px;
  height: 20px;
}

.preview-tile.small {
  width: 5px;
  height: 5px;
}

.gb-green {
  background-color: #6b790e;
}

.gb-grey {
  background-color: #c5bfbf;
}

.gb-grey-dark {
  background-color: #5a5760;
}

.gb-white {
  background-color: #fff;
}

.gb-black {
  background-color: #000;
}
