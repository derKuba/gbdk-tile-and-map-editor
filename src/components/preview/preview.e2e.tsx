import { newE2EPage } from '@stencil/core/testing';

describe('mvjp-preview', () => {
    it('renders', async () => {
        const page = await newE2EPage();
        await page.setContent('<mvjp-preview></mvjp-preview>');

        const element = await page.find('mvjp-preview');
        expect(element).toHaveClass('hydrated');
    });
});
