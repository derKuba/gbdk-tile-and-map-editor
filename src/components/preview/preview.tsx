import { Component, Host, h, Prop } from '@stencil/core';

import Tunnel from '../../provider/StateProvider';

@Component({
    tag: 'mvjp-preview',
    styleUrl: 'preview.css',
    shadow: true,
})
export class Preview {
    /**
     *   which format
     */
    @Prop() size: string;

    /**
     *   global matrix
     */
    @Prop() matrix: number[][][];

    /**
     *   chosen matrix tile
     */
    @Prop() indexTileMatrix: number;

    /**
     *   usage without a list
     */
    @Prop() standalone: boolean = false;

    private mapColorCodeToColour = colorCode => {
        const colors: string[] = ['white', 'grey', 'grey-dark', 'black'];
        return colors[colorCode];
    };

    render() {
        const sizeClass = this.size === 's' ? 'small' : this.size === 'l' ? 'large' : null;
        const tileMatrix = this.matrix[this.indexTileMatrix];

        const previewField = tileMatrix.map((tileRow: number[], y: number) => (
            <div class="matrixRow" key={`index_${y}`}>
                {tileRow.map((colorCode: number, x: number) => (
                    <div
                        class={`preview-tile ${sizeClass} gb-${this.mapColorCodeToColour(colorCode)}`}
                        key={`index_${y}_${x}`}></div>
                ))}
            </div>
        ));

        return (
            <Host>
                <div class={`preview-box ${sizeClass} ${this.standalone ? 'standalone' : ''}`}>{previewField}</div>
            </Host>
        );
    }
}

Tunnel.injectProps(Preview, ['matrix', 'indexTileMatrix']); // eslint-disable-line @stencil/ban-side-effects
