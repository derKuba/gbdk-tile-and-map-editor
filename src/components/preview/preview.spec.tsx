import { newSpecPage } from '@stencil/core/dist/testing';
import { h } from '@stencil/core';

import { Preview } from './preview';

describe('mvjp-preview', () => {
    it('builds', () => {
        expect(new Preview()).toBeTruthy();
    });

    it('matches snapshot', async () => {
        let matrix: number[][][] = [];
        for (let i = 0; i < 12; i++) {
            matrix.push(Array.from({ length: 8 }, () => Array.from({ length: 8 }, () => 0)));
        }

        matrix[0] = [
            [1, 1, 1, 0, 1, 1, 1, 0],
            [0, 0, 1, 0, 1, 0, 0, 1],
            [0, 0, 1, 0, 1, 1, 1, 0],
            [0, 0, 1, 0, 1, 0, 0, 0],
            [0, 1, 1, 0, 1, 0, 0, 0],
            [1, 0, 1, 0, 1, 0, 0, 0],
            [0, 1, 0, 0, 1, 0, 0, 0],
            [0, 0, 0, 0, 0, 0, 0, 0],
        ];

        const page = await newSpecPage({
            components: [Preview],
            template: () => {
                return (
                    <div>
                        <mvjp-preview size="s" matrix={matrix} indexTileMatrix={0} />

                        <mvjp-preview size="l" matrix={matrix} indexTileMatrix={0} />
                    </div>
                );
            },
        });

        expect(page.root).toMatchSnapshot();
    });
});
