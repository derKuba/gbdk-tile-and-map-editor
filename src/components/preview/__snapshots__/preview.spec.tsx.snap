// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mvjp-preview matches snapshot 1`] = `
<mvjp-preview>
  <mock:shadow-root>
    <div class="preview-box small">
      <div class="matrixRow">
        <div class="gb-grey preview-tile small"></div>
        <div class="gb-grey preview-tile small"></div>
        <div class="gb-grey preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
        <div class="gb-grey preview-tile small"></div>
        <div class="gb-grey preview-tile small"></div>
        <div class="gb-grey preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
      </div>
      <div class="matrixRow">
        <div class="gb-white preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
        <div class="gb-grey preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
        <div class="gb-grey preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
        <div class="gb-grey preview-tile small"></div>
      </div>
      <div class="matrixRow">
        <div class="gb-white preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
        <div class="gb-grey preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
        <div class="gb-grey preview-tile small"></div>
        <div class="gb-grey preview-tile small"></div>
        <div class="gb-grey preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
      </div>
      <div class="matrixRow">
        <div class="gb-white preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
        <div class="gb-grey preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
        <div class="gb-grey preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
      </div>
      <div class="matrixRow">
        <div class="gb-white preview-tile small"></div>
        <div class="gb-grey preview-tile small"></div>
        <div class="gb-grey preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
        <div class="gb-grey preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
      </div>
      <div class="matrixRow">
        <div class="gb-grey preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
        <div class="gb-grey preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
        <div class="gb-grey preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
      </div>
      <div class="matrixRow">
        <div class="gb-white preview-tile small"></div>
        <div class="gb-grey preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
        <div class="gb-grey preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
      </div>
      <div class="matrixRow">
        <div class="gb-white preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
        <div class="gb-white preview-tile small"></div>
      </div>
    </div>
  </mock:shadow-root>
</mvjp-preview>
`;
